jenkins/bootstrap.py

```diff
@@ -724,26 +728,52 @@ def gubernator_uri(paths):
     return job
 
 
-def setup_root(call, root, repo, branch, pull, ssh, git_cache, clean):
+def setup_root(call, root, repos, branches, pulls, ssh, git_cache, clean):
     """Create root dir, checkout repo and cd into resulting dir."""
-    logging.info(
-        'Check out %s at %s...',
-        os.path.join(root, repo or ''),
-        pull if pull else branch)
     if not os.path.exists(root):
         os.makedirs(root)
-    os.chdir(root)
-    if repo:
-        checkout(call, repo, branch, pull, ssh, git_cache, clean)
-    elif branch or pull:
-        raise ValueError('--branch and --pull require --repo', branch, pull)
-
-
-def bootstrap(
-    job, repo, branch, pull, root, upload, robot, timeout=0,
-    use_json=False, ssh=False, git_cache='', clean=False):
+    root_dir = os.path.realpath(root)
+    logging.info('Root: %s', root_dir)
+    if repos:
+        if branches:
+            if len(branches) != len(repos):
+                raise ValueError('--branch != --repo count', repos, branches)
+        if pulls:
+            if len(pulls) != len(repos):
+                raise ValueError('--pull != --repo count', repos, pulls)
+        for num, repo in enumerate(repos):
```

Comment:
this might be cleaner as `for repo, branch, pull in itertools.izip_longest(repos, branches, pulls):`